modules/tensor_mechanics/src/bcs/CoupledPressureBC.C

```diff
@@ -0,0 +1,39 @@
+//* This file is part of the MOOSE framework
```

Comment:
Does this need to be a separate class from the regular Pressure class? We're already doing a couple of switches between different ways of applying it in there anyway?

Reply:
`Pressure` has that `factor`, `function` and `postprocessor` stuff which IMO does not make sense in this context. Again, the pressure variable will be computed by an external code in 95% of cases and transferred into.

Reply:
Also, I personally prefer classes that do just one thing rather then a lot of things

Reply:
Sure, I understand that. However, the Pressure class already does multiple things, and this would be a minor addition. 

We probably don't actually need `factor` and `function` because you can prescribe a constant value for `function`. The same is true for the coupled variable. I'd just like to discuss this for a minute.

Reply:
Turns out I'm not going to be around today. I see your point, but I don't think that pattern is followed all that much actually, and we would have a massive number of classes if we actually did follow it everywhere. I'm fine with how you have it, but I'd prefer we go all one way or the other. Following this line of thinking, it seems like we should have a separate Action and syntax block for CoupledPressure as well...

Reply:
I am fine with doing a different syntax. That's cleaner than having it all in one action.

Something like?
```
[BCs]
  [./CoupledPressure]
    displacements = '...'
    boundary = '<bnd>'
    coupled_pressure = <variable name>
  [../]
[]
```

Here, it looks like `coupled_` prefix is redundant, because it is in the block name. I do not want to go with `variable`, because that could be misinterpreted as the variable this is applied to. What do you think?

Reply:
Sure, if you bust this out into its own syntax block, then I think `pressure` is better than `coupled_pressure`.

Reply:
Same goes for the BC itself.